pkg/app/pipectl/prompt/reader.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prompt
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	maximum_retry = 5 // maximum number of retry for required fields
+)
+
+// Reader is an interface that reads input from user or mock.
+type Reader interface {
+	// ReadString reads a string with showing message. If the input is empty, it returns an empty string.
+	// If the input is more than one word, it returns an error.
+	ReadString(message string) (string, error)
+	// ReadStrings reads a string slice separated by blank space with showing message. If the input is empty, it returns an empty slice.
+	ReadStrings(message string) ([]string, error)
+	// ReadInt reads an integer with showing message. If the input is empty, it returns 0.
+	// If the input is not an integer, it returns an error.
+	ReadInt(message string) (int, error)
+	// ReadStringRequired reads a string with showing message. If the input is empty, it asks again.
+	// If number of retry exceeds maximum_retry, it returns an error.
+	// If the input is more than one word, it returns an error.
+	ReadStringRequired(message string) (string, error)
+}
```

Comment:
[ask] What's the purpose of the Interface? 
I think reading data from stdin is a concrete usecase, so it might not be necessary abstraction 👀 
WDYT?

Reply:
The purpose is to use mock in `initialize/ecs_test.go`. 
In tests, stdin is difficult to use but we need to pass values instead of users.
So in tests I use `mockReader`(in `prompt/reader_mock.go`), which implements `Reader`.

Reply:
Thx, I got it!
So you can create `NewReader(input io.Reader)` and mock the `input` 👍
You can create the implementation of `io.Reader` from a string using `strings.NewReader` https://pkg.go.dev/strings#NewReader

Like this ↓
```
// default
r := prompt.NewReader(os.Stdin)

// when test
mockStr := `foo
foo bar
`
mockStdin := strings.NewReader(mockStr)
r := prompt.NewReader(mockStdin)
```

Reply:
@ffjlabo 
great, amazing, thank you so much!
It worked well, and I could remove the `Reader` interface!
`strings.NewReader` is what i wanted.